trin-bridge/src/bridge.rs

```diff
@@ -0,0 +1,580 @@
+                    Ok(val) => val,
+                    Err(msg) => {
+                        warn!("error getting headers, skipping iteration: {msg:?}");
+                        block_index = latest_block + 1;
+                        sleep(Duration::from_secs(LATEST_BLOCK_POLL_RATE)).await;
+                        continue;
+                    }
+                };
+                // epoch index is 0 here bc it's irrelevant for post-merge headers
+                let mut gossip_batch = GossipBatch::new(block_range_to_gossip, 0, full_headers);
+                if let Err(msg) = self.gossip_headers(&mut gossip_batch).await {
+                    warn!("error gossiping headers, skipping iteration: {msg:?}");
+                    block_index = latest_block + 1;
+                    sleep(Duration::from_secs(LATEST_BLOCK_POLL_RATE)).await;
+                    continue;
+                };
+                // Sleep for 5 seconds to allow headers to saturate network,
+                // since they must be available for body / receipt validation.
+                sleep(Duration::from_secs(HEADER_SATURATION_DELAY)).await;
+                if let Err(err) = self.serve_bodies(&mut gossip_batch).await {
+                    warn!("error gossiping bodies: {err:?}");
+                };
+                if let Err(err) = self.serve_receipts(&mut gossip_batch).await {
+                    warn!("error gossiping receipts: {err:?}");
+                };
+                gossip_batch.display_stats();
+                block_index = gossip_batch.range.end;
+            }
+            sleep(Duration::from_secs(LATEST_BLOCK_POLL_RATE)).await;
+        }
+    }
+
+    pub async fn launch_backfill(&self, starting_epoch: Option<u64>) {
+        let latest_block = get_latest_block_number().await.expect(
+            "Error launching bridge in backfill mode. Unable to get latest block from provider.",
+        );
+        let mut epoch_index = match starting_epoch {
+            Some(val) => {
+                if val * EPOCH_SIZE > latest_block {
+                    panic!(
+                        "Starting epoch is greater than current epoch. 
+                        Please specify a starting epoch that begins before the current block."
+                    );
+                }
+                val
+            }
+            None => 0,
+        };
+        let current_epoch = latest_block / EPOCH_SIZE;
+        while epoch_index < current_epoch {
+            let start = epoch_index * EPOCH_SIZE;
+            let end = start + EPOCH_SIZE;
+            // Using epoch_size chunks & epoch boundaries ensures that every
+            // "chunk" shares an epoch accumulator avoiding the need to
+            // look up the epoch acc on a header by header basis
+            let block_range_to_gossip = Range { start, end };
+            let full_headers = match Bridge::get_headers(&block_range_to_gossip).await {
```

Comment:
Imo, leaving this as a todo for a future pr is sufficient, since we're not really using backfill atm & getting this into master is very helpful for devops stuff. But as I've started using backfill a bit I've noticed that this design is not great since if **any** gossip batch request to geth within the range fails, then we stop and move onto the next range. Obvs, this isn't ideal, since 63/64 batched header requests to geth could succeed, and if only one fails, then we essentially drop the data and move onto the next range. Instead, we should use an option here so that we can proceed with gossiping hwp/bodies/receipts for all complete gossip batches.